app/models/user.rb

```diff
@@ -159,4 +160,10 @@ def blocked_or_hidden_posts(keyword, post_user_ids, full_user_ids)
       post_ids
     end
   end
+
+  def update_flat_posts
+    return unless saved_change_to_username || saved_change_to_deleted
+    post_ids = (Post.where(user_id: id).pluck(:id) + Reply.where(user_id: id).select(:post_id).distinct.pluck(:post_id)).uniq
```

Comment:
Possibly faster to use PostAuthor here

Reply:
-- oh, yeah, PostAuthor probably would cover that